rustls/src/error.rs

```diff
@@ -47,6 +47,9 @@ pub enum Error {
     /// The certificate verifier doesn't support the given type of name.
     UnsupportedNameType,
 
+    /// We couldn't retrieve a public key.
+    UnsupportedPublicKey,
```

Comment:
bike-shedding: I wonder if there's a better name for this since it reflects that we couldn't retrieve a public key more than that the public key was in-hand, but unsupported. `UnknownPublicKey`?

Reply:
In the spirit of bike-shedding: would it be appropriate to use `InconsistentKeys::Unknown`?

Reply:
That feels reasonable to me, but if we go this direction we should update the part of the docs that say "Specific failure cases from [keys_match](https://docs.rs/rustls/latest/rustls/sign/struct.CertifiedKey.html#method.keys_match).".